File: hospital-management-frontend/src/services/doctorService.js
import api from './api'

const doctorService = {
  getAllDoctors: async () => {
    const response = await api.get('/doctors')
    return response.data
  },

  getDoctorById: async (id) => {
    const response = await api.get(`/doctors/${id}`)
    return response.data
  },

  getDoctorByUserId: async (userId) => {
    const response = await api.get(`/doctors/user/${userId}`)
    return response.data
  },

  getDoctorsBySpecialization: async (specialization) => {
    const response = await api.get(`/doctors/specialization/${specialization}`)
    return response.data
  },

  updateDoctor: async (id, doctorData) => {
    const response = await api.put(`/doctors/${id}`, doctorData)
    return response.data
  },

  deleteDoctor: async (id) => {
    const response = await api.delete(`/doctors/${id}`)
    return response.data
  },
}

export default doctorService

File: hospital-app/src/main/java/com/example/hospital/service/DoctorService.java
package com.example.hospital.service;

import com.example.hospital.entity.Doctor;
import com.example.hospital.repository.DoctorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DoctorService {

    @Autowired
    private DoctorRepository doctorRepository;

    // Get all doctors
    public List<Doctor> getAllDoctors() {
        return doctorRepository.findAll();
    }

    // Get doctor by ID
    public Optional<Doctor> getDoctorById(Long id) {
        return doctorRepository.findById(id);
    }

    // Get doctor by user ID
    public Optional<Doctor> getDoctorByUserId(Long userId) {
        return doctorRepository.findByUserId(userId);
    }

    // Get doctors by specialization
    public List<Doctor> getDoctorsBySpecialization(String specialization) {
        return doctorRepository.findBySpecialization(specialization);
    }

    // Update doctor
    public Doctor updateDoctor(Long id, Doctor doctorDetails) {
        Optional<Doctor> doctorOpt = doctorRepository.findById(id);
        
        if (doctorOpt.isPresent()) {
            Doctor doctor = doctorOpt.get();
            doctor.setSpecialization(doctorDetails.getSpecialization());
            doctor.setQualification(doctorDetails.getQualification());
            doctor.setExperience(doctorDetails.getExperience());
            doctor.setAvailableSlots(doctorDetails.getAvailableSlots());
            return doctorRepository.save(doctor);
        }
        
        return null;
    }

    // Delete doctor
    public boolean deleteDoctor(Long id) {
        if (doctorRepository.existsById(id)) {
            doctorRepository.deleteById(id);
            return true;
        }
        return false;
    }
}